compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/window/Popup.skiko.kt

```diff
@@ -419,17 +419,26 @@ private fun PopupLayout(
     onOutsidePointerEvent: ((eventType: PointerEventType, button: PointerButton?) -> Unit)? = null,
     content: @Composable () -> Unit
 ) {
+    // Use a MutableState directly to avoid recomposing when the value changes
+    val parentBoundsInWindow: MutableState<IntRect> = remember { mutableStateOf(IntRect.Zero) }
+    EmptyLayout(Modifier.onPlaced { childCoordinates ->
+        // This will be called before the popup measure policy is actually asked to calculate
+        // the popup's position, so it will never see the initial value of IntRect.Zero
+        childCoordinates.parentCoordinates?.let {
+            val layoutPosition = it.positionInWindow().round()
+            val layoutSize = it.size
+            parentBoundsInWindow.value = IntRect(layoutPosition, layoutSize)
+        }
+    })
+
     val currentContent by rememberUpdatedState(content)
-    var layoutParentBoundsInWindow: IntRect? by remember { mutableStateOf(null) }
-    EmptyLayout(Modifier.parentBoundsInWindow { layoutParentBoundsInWindow = it })
     val layer = rememberComposeSceneLayer(
         focusable = properties.focusable
     )
     layer.setKeyEventListener(onPreviewKeyEvent, onKeyEvent)
     layer.setOutsidePointerEventListener(onOutsidePointerEvent)
     layer.Content {
         val platformInsets = properties.platformInsets
-        val parentBoundsInWindow = layoutParentBoundsInWindow ?: return@Content
```

Comment:
Can we keep this logic? Or add require/assert to make sure that it's initialized? I mean for possible future changes

Reply:
I'd need to make the type nullable (`IntRect?`) again, which I prefer not to do.

But the unit test I added checks exactly this...